Physical_Engine/Create/FramingElement.cs

```diff
@@ -0,0 +1,98 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Physical.Elements;
+using BH.oM.Physical.FramingProperties;
+using BH.oM.Geometry;
+
+
+namespace BH.Engine.Physical
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates a physical Beam element. To generate elements compatible with structural packages, have a look at the Bar class")]
+        [Input("location", "The centre line geometry of the beam")]
+        [Output("Beam", "The created physical beam")]
+        public static Beam Beam(ICurve location, IFramingElementProperty property, string name = "")
```

Comment:
Thanks @ZiolkowskiJakub . I Intentionally left it out as I did not think adding "The name of the beam" would give more information to the user, and to avoid more clutter, but can add it if dont agree?

Reply:
Hi @IsakNaslundBh,

Thanks for clarification. Your approach sounds good to me. It builds correctly on my machine so approval from me.

Reply:
I'm going to play uninformed user here and ask why does the creation of the beam need to know my name? Am I assigning my name to the beam to say "this beam was made by @FraserGreenroyd "?

As much as I would agree with you @IsakNaslundBh that it should be obvious what the `name` input is referring to, erring on the side of an uninformed user it might not be as obvious as we would think.

That said, I don't think it needs to be
> The name of the beam as an alphanumeric string that can identify the beam to its family and friends

But something which clarifies to the user they are indeed naming the beam rather than doing something else with it is probably worthwhile.

See also: 
![image](https://user-images.githubusercontent.com/18049174/59609995-631f2300-9110-11e9-978f-f7214d035d95.png)
https://github.com/BHoM/BHoM_Engine/blob/master/Environment_Engine/Create/Panel.cs